torchinductor/codegen/cpp_prefix.h

```diff
@@ -3,7 +3,9 @@
 #include <cmath>
 #include <cstdlib>
 #include <limits>
+#ifdef _OPENMP
 #include <omp.h>
+#endif
```

Comment:
Not blocking at all, just worth considering. This is safe because we only use `#pragma omp` from omp. If we ever introduce a direct call into an omp.h header, that call will fail to compile. As we don't have osx CI, that could go unnoticed.

A few potential solutions, not for this PR:

A) Add OSX CI
B) have an omp_available bool in cpp.py to guard omp calls
C) Instruct OSX / provide infra for users to install libomp.